From 7f9d3dc3ddf41cfaeeedf83f669c852032f29549 Mon Sep 17 00:00:00 2001
From: Petr Vobornik <pvoborni@redhat.com>
Date: Fri, 2 Nov 2012 14:02:39 +0100
Subject: [PATCH] Web UI: disable global forwarding per zone

Web UI part of 'disable global forwaring per zone' effort.

Option "Forwarding disabled" was added to 'DNS global config' and 'DNS zone' forwarding policy. It corresponds to 'none' value of idnsforwardpolicy.

https://fedorahosted.org/freeipa/ticket/3209
---
 install/ui/dns.js                  | 8 ++++++++
 install/ui/test/data/ipa_init.json | 1 +
 ipalib/plugins/internal.py         | 1 +
 3 files changed, 10 insertions(+)

diff --git a/install/ui/dns.js b/install/ui/dns.js
index 43703e03f3e6dc4061f52d1f865db85b0e9c8502..fd6127d3467a3f89aa013e55535c918ae3e383c9 100644
--- a/install/ui/dns.js
+++ b/install/ui/dns.js
@@ -72,6 +72,10 @@ IPA.dns.config_entity = function(spec) {
                                 {
                                     value: 'only',
                                     label: IPA.messages.objects.dnsconfig.forward_only
+                                },
+                                {
+                                    value: 'none',
+                                    label: IPA.messages.objects.dnsconfig.forward_none
                                 }
                             ]
                         },
@@ -201,6 +205,10 @@ IPA.dns.zone_entity = function(spec) {
                             {
                                 value: 'only',
                                 label: IPA.messages.objects.dnsconfig.forward_only
+                            },
+                            {
+                                    value: 'none',
+                                    label: IPA.messages.objects.dnsconfig.forward_none
                             }
                         ]
                     },
diff --git a/install/ui/test/data/ipa_init.json b/install/ui/test/data/ipa_init.json
index 2b95c91d314bebb55ec57d845af0f003a154f617..41f2c6270c746edd97fab960a35c6a6acae7d6da 100644
--- a/install/ui/test/data/ipa_init.json
+++ b/install/ui/test/data/ipa_init.json
@@ -222,6 +222,7 @@
                         "delegation": {},
                         "dnsconfig": {
                             "forward_first": "Forward first",
+                            "forward_none": "Forwarding disabled",
                             "forward_only": "Forward only",
                             "options": "Options"
                         },
diff --git a/ipalib/plugins/internal.py b/ipalib/plugins/internal.py
index 6195ddd930a59336899293e272991bc5668322b0..cfb5d60f9f8a9f748b58dd8364a8690bf6724755 100644
--- a/ipalib/plugins/internal.py
+++ b/ipalib/plugins/internal.py
@@ -359,6 +359,7 @@ class i18n_messages(Command):
             },
             "dnsconfig": {
                 "forward_first": _("Forward first"),
+                "forward_none": _("Forwarding disabled"),
                 "forward_only": _("Forward only"),
                 "options": _("Options"),
             },
-- 
1.7.11.7

